boto/s3/connection.py

```diff
@@ -89,9 +89,8 @@ def build_host(self, server, bucket):
             return self.get_bucket_server(server, bucket)
 
     def build_auth_path(self, bucket, key=''):
+        bucket = six.ensure_text(bucket, encoding='utf-8')
         key = get_utf8able_str(key)
-        if isinstance(bucket, bytes):
-            bucket = bucket.decode('utf-8')
         path = ''
```

Comment:
Let's make this explicitly unicode, i.e. `path = u''`, so that everything appended to it afterward is coerced to unicode (mainly concerned with py2 code branches, and coding defensively for if/when someone adds another branch).

Reply:
That sounds smart! More explicit is better in this case imo